API/PROBLEMS/abuse/of_type_aliasing

```diff
@@ -0,0 +1,43 @@
+Frequently (in the Coq codebase) type aliases are used as a way to avoid doing something
+that requires more effort (when writing the code) but unfortunatelly also means that
+it makes the code more opaque (to the reader).
+
+EXAMPLE:
+
+Instead of doing
+
+   type foo = bool
+or
+   type foo = int
+or
+   ...
+
+we should consider using labeled function arguments
+(where values of these types are applied)
+
+EXAMPLE
+
+E.g., in "Impargs" module, we have this:
+
+        type locality = bool
+        type polymorphic = bool
+        type definition_kind = locality * polymorphic * definition_object_kind
+
+The is perhaps better than:
+
+        type definition_kind = bool * bool * definition_object_kind
+
+but still far from ideal. We are abusing type-aliasing to avoid using records. Why?
+Why not simply define the same type as:
+
+	type definition_kind = {locality : bool;
+                                polymorphic : bool;
+                                definition_object_kind : definition_object_kind}
+
+We would not only make clear the meaning of individual fields in the type definition itself,
+but we would at the same time ensure that the meaning of the corresponding values is clear.
```

Comment:
For this we can define types isomorphic to bool, eg `type locality = Local | NotLocal`.